clients/admin-ui/src/features/privacy-requests/configuration/MailgunEmailConfiguration.tsx

```diff
@@ -0,0 +1,223 @@
+import { Box, Button, Divider, Heading, Stack } from "@fidesui/react";
+import { Form, Formik } from "formik";
+import { useState } from "react";
+
+import { CustomTextInput } from "~/features/common/form/inputs";
+import { getErrorMessage } from "~/features/common/helpers";
+import { useAlert } from "~/features/common/hooks";
+import {
+  useCreateMessagingConfigurationMutation,
+  useCreateMessagingConfigurationSecretsMutation,
+  // useGetMessagingConfigurationDetailsQuery,
+} from "~/features/privacy-requests/privacy-requests.slice";
+
+const MailgunEmailConfiguration = () => {
+  const { errorAlert, successAlert } = useAlert();
+  const [configurationStep, setConfigurationStep] = useState("");
+  // const { data: messagingDetails } = useGetMessagingConfigurationDetailsQuery({
+  //   type: "mailgun",
+  // });
+  const [createMessagingConfiguration] =
+    useCreateMessagingConfigurationMutation();
+  const [createMessagingConfigurationSecrets] =
+    useCreateMessagingConfigurationSecretsMutation();
+
+  const handleMailgunConfiguration = async (value: { domain: string }) => {
+    const payload = await createMessagingConfiguration({
+      type: "mailgun",
+      details: {
+        is_eu_domain: "false",
+        domain: value.domain,
+      },
+    });
+
+    if ("error" in payload) {
+      errorAlert(
+        getErrorMessage(payload.error),
+        `Configuring storage type has failed to save due to the following:`
+      );
+    } else {
+      successAlert(`Configure storage type saved successfully.`);
+      setConfigurationStep("2");
+    }
+  };
+
+  const handleMailgunAPIKeyConfiguration = async (value: {
+    api_key: string;
+  }) => {
+    const payload = await createMessagingConfigurationSecrets({
+      mailgun_api_key: value.api_key,
+    });
+
+    if ("error" in payload) {
+      errorAlert(
+        getErrorMessage(payload.error),
+        `Configuring storage type has failed to save due to the following:`
+      );
+    } else {
+      successAlert(`Configure storage type saved successfully.`);
+      setConfigurationStep("3");
+    }
+  };
+
+  // const handleTestConnection = () => {
+  //   setConfigurationStep("4");
+  // };
+
+  const MAILGUN_MESSAGING_CONFIG_FORM_ID = "mailgun-messaging-config-form-id";
+  const MAILGUN_MESSAGING_CONFIG_API_KEY_FORM_ID =
+    "mailgun-messaging-config-api-key-form-id";
```

Comment:
Extract constants outside of the component function (above, within the same file is ok).